src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -329,15 +320,68 @@ void EIRPlantLoopHeatPump::doPhysics(EnergyPlusData &state, Real64 currentLoad)
         return;
     }
 
+    // dispatch to specific physics calculations based on the heat pump type
+    if (this->waterSource) {
+        this->doPhysicsWSHP(state, currentLoad);
+    } else if (this->airSource) {
+        this->doPhysicsASHP(state, currentLoad);
+    }
```

Comment:
Now we have two separate functions `doPhysicsWSHP()` and `doPhysicsASHP()` that are wrapped in existing `doPhysics()`. The `doPhysics()` is a virtual function used for `EIRFuelFiredHeatPump` module as well.